tests/main/snapd-snap/task.yaml

```diff
@@ -1,29 +1,107 @@
 summary: Ensure snapd builds as a snap
 
-# This is what the snapcraft builder on launchpad uses.
-systems: [ubuntu-16.04-64]
+# TODO: we don't need to run this test on as many systems when we have a way to
```

Comment:
We should be able to use upload-artifact and download-artifact actions to do that. A simple curl on the command line does not work last time I checked as one needs to be logged in to github (or at least present a valid token). I suppose uploading a repacked snapd snap with spread project would simplify prepare as we could conditionally skip repacking the snapd snap too?

Reply:
Yes I've looked into this a bit but as you say you need a GitHub token in order to download it. I don't think we should block this PR on trying to do that, but it would be really great to be able to do that some day